vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1320,6 +1327,46 @@ def execute_model(
                                             all_gather_group=get_tp_group())
             logits = None
         else:
+            if self.input_batch.pooling_params:
+                assert self.input_batch.num_reqs ==\
+                    len(self.input_batch.pooling_params), \
+                "Either all or none of the requests in" \
+                " a batch must be pooling request"
+
+                total_len = num_scheduled_tokens_np.sum()
+                extracted_hidden_states = list(
+                    torch.split(hidden_states[:total_len],
+                                num_scheduled_tokens_np.tolist()))
+
+                pooling_metadata = self.input_batch.pooling_metadata
+
+                raw_pooler_output = self.model.pooler(
+                    hidden_states=extracted_hidden_states,
+                    pooling_metadata=pooling_metadata)
+
+                pooler_output: list[Optional[torch.Tensor]] = []
+                seq_lens = self.seq_lens[:self.input_batch.num_reqs]
+                for raw_output, seq_len, prompt_len in zip(
+                        raw_pooler_output, seq_lens,
+                        pooling_metadata.prompt_lens):
+
+                    if seq_len == prompt_len:
+                        pooler_output.append(raw_output.data.to("cpu"))
```

Comment:
Why not just `cpu()`?
```suggestion
                        pooler_output.append(raw_output.cpu())
```

Reply:
The .data is necessary, but cpu() is better than to("cpu").